gobblin-cluster/src/main/java/gobblin/cluster/GobblinClusterUtils.java

```diff
@@ -57,4 +64,52 @@ public static Path getAppWorkDirPath(FileSystem fs, String applicationName, Stri
   public static String getAppWorkDirPath(String applicationName, String applicationId) {
     return applicationName + Path.SEPARATOR + applicationId;
   }
+
+  /**
+   * state stores used for storing work units and task states
+   */
+  public static class StateStores {
+    public final StateStore<TaskState> taskStateStore;
+    public final StateStore<WorkUnit> wuStateStore;
+    public final StateStore<MultiWorkUnit> mwuStateStore;
```

Comment:
This is due to the deserialization. The extension in the state name of wu or mwu indicates which type to deserialize to and this logic is external to the StateStore.